### ISSUE TITLE
Community themes contain resources for Commerce edition modules

### ISSUE THREAD

**Issue by clockworkgeek** [magento/magento2#21446]
### Summary (*)

The three standard themes have several folders which reference modules that do not exist in Community edition.  They mostly contain `_module.less` files and this leads to unused style rules in the deployed CSS.  Also it's a little bit annoying.

### Examples (*)

The following folders are superfluous:

    adminhtml/Magento/backend/Magento_AdvancedCheckout
    adminhtml/Magento/backend/Magento_Banner
    adminhtml/Magento/backend/Magento_CatalogPermissions
    adminhtml/Magento/backend/Magento_CustomerBalance
    adminhtml/Magento/backend/Magento_GiftCard
    adminhtml/Magento/backend/Magento_GiftRegistry
    adminhtml/Magento/backend/Magento_GiftWrapping
    adminhtml/Magento/backend/Magento_Reward
    adminhtml/Magento/backend/Magento_Rma
    adminhtml/Magento/backend/Magento_VersionsCms
    adminhtml/Magento/backend/Magento_VisualMerchandiser
    frontend/Magento/black/Magento_AdvancedCheckout
    frontend/Magento/black/Magento_Banner
    frontend/Magento/black/Magento_CatalogEvent
    frontend/Magento/black/Magento_GiftCard
    frontend/Magento/black/Magento_GiftCardAccount
    frontend/Magento/black/Magento_GiftRegistry
    frontend/Magento/black/Magento_GiftWrapping
    frontend/Magento/black/Magento_Invitation
    frontend/Magento/black/Magento_MultipleWishlist
    frontend/Magento/black/Magento_Rma
    frontend/Magento/black/Magento_VersionsCms
    frontend/Magento/luma/Magento_AdvancedCheckout
    frontend/Magento/luma/Magento_CustomerBalance
    frontend/Magento/luma/Magento_GiftCard
    frontend/Magento/luma/Magento_GiftCardAccount
    frontend/Magento/luma/Magento_GiftWrapping
    frontend/Magento/luma/Magento_Invitation
    frontend/Magento/luma/Magento_MultipleWishlist
    frontend/Magento/luma/Magento_Reward
    frontend/Magento/luma/Magento_Rma

### Proposed solution

Move the mentioned resources to the appropriate theme or the modules' own `view` folders.

**Comment by magento-engcom-team** [comment]
Hi @clockworkgeek. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@clockworkgeek do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @clockworkgeek thank you for you report, this modules not deployed to frontend or adminhtml, if you recheck `pub/static` folder this files not exists.
![deepinscreenshot_select-area_20190226095518](https://user-images.githubusercontent.com/40756439/53396262-b4ae7100-39ac-11e9-8717-86500ab26896.png)
![deepinscreenshot_select-area_20190226095546](https://user-images.githubusercontent.com/40756439/53396264-b4ae7100-39ac-11e9-8da6-19cbb0fd6ffa.png)

**Comment by clockworkgeek** [comment]
Thank you @engcom-backlog-nazar for looking into this.  LESS files are not static so will not be deployed to `pub/static`.  They are copied to `var/view_preprocessed` and then compiled into `styles.css`, `styles-m.css`, and `styles-l.css`.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98455`, `MAGETWO-98456` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by sdzhepa** [comment]
Hello @clockworkgeek 

Thank you for contribution and collaboration!

I can confirm that from the first glance there is no sense to have these *.less files in Open Source Codebase.
And it is logical and obvious decision to move them to Commerce Source.

But it will have negative consequences on composer installation approach. Especially how Magento builds the packages for composer
We do not have separate theme packages(blank/luma) for Open Source and Commerce yet

To develop complex fix for this issue need to have access to "Magento Composer package builder" which is internal only.

So, unfortunately, we have to close this PR and related issue and wait for fix from Magento Core

Also, want to admit that the internal team several times tried to fix this issue by moving *.less but changes were reverted.

**Additional information**
Internal Jira ticket with this issue: 
-  `MAGETWO-50837` "Remove unused less files from backend folders"

Currently has `REOPEN` status and  blocked by another architecture ticket:
-  `MAGETWO-44582` "Theme Extension Strategy"

**Comment by clockworkgeek** [comment]
Thank you @sdzhepa for looking into this in detail. I hope you will find a solution.

**Comment by andrewbess** [comment]
Hello @sdzhepa 
I think we can move the `less` files of `adminhtml` area from Magento CE to Magento EE (into modules).
Sergey, what do you think about it?

**Comment by sdzhepa** [comment]
Hello @andrewbess 

It seems your proposition has sense. 

I see that internal Jira ticket  `MAGETWO-44582`  "Theme Extension Strategy" still in status = OPEN
Let's try to ask @antonkril to help with this issue or with involving appropriate contact to discuss it here.

**Comment by antonkril** [comment]
Makes sense. @maghamed @buskamuza, can you investigate the reason why the previous attempts failed?

**Comment by andrewbess** [comment]
@antonkril 
I think it because it doesn't look like the correct solution.
Please take a look at the screenshot below.
<img width="1009" alt="wrong-file-path" src="https://user-images.githubusercontent.com/10788564/78464796-3e735680-76f6-11ea-9326-0e4c346a89e8.png">

**Comment by antonkril** [comment]
Yeah, I remembered that discussion now. In both options, we end up with files that are potentially unused. But your proposal was about adminhtml, which, with AdminUI modules should not be a problem.

**Comment by andrewbess** [comment]
So, I add the checker of the fixed file here

- [ ] adminhtml/Magento/backend/Magento_AdvancedCheckout
- [ ] adminhtml/Magento/backend/Magento_Banner
- [ ] adminhtml/Magento/backend/Magento_CatalogPermissions
- [ ] adminhtml/Magento/backend/Magento_CustomerBalance
- [ ] adminhtml/Magento/backend/Magento_GiftCard
- [ ] adminhtml/Magento/backend/Magento_GiftRegistry
- [ ] adminhtml/Magento/backend/Magento_GiftWrapping
- [ ] adminhtml/Magento/backend/Magento_Reward
- [ ] adminhtml/Magento/backend/Magento_Rma
- [ ] adminhtml/Magento/backend/Magento_VersionsCms
- [ ] adminhtml/Magento/backend/Magento_VisualMerchandiser
- [ ] frontend/Magento/black/Magento_AdvancedCheckout
- [ ] frontend/Magento/black/Magento_Banner
- [ ] frontend/Magento/black/Magento_CatalogEvent
- [ ] frontend/Magento/black/Magento_GiftCard
- [ ] frontend/Magento/black/Magento_GiftCardAccount
- [ ] frontend/Magento/black/Magento_GiftRegistry
- [ ] frontend/Magento/black/Magento_GiftWrapping
- [ ] frontend/Magento/black/Magento_Invitation
- [ ] frontend/Magento/black/Magento_MultipleWishlist
- [ ] frontend/Magento/black/Magento_Rma
- [ ] frontend/Magento/black/Magento_VersionsCms
- [ ] frontend/Magento/luma/Magento_AdvancedCheckout
- [ ] frontend/Magento/luma/Magento_CustomerBalance
- [ ] frontend/Magento/luma/Magento_GiftCard
- [ ] frontend/Magento/luma/Magento_GiftCardAccount
- [ ] frontend/Magento/luma/Magento_GiftWrapping
- [ ] frontend/Magento/luma/Magento_Invitation
- [ ] frontend/Magento/luma/Magento_MultipleWishlist
- [ ] frontend/Magento/luma/Magento_Reward
- [ ] frontend/Magento/luma/Magento_Rma

**Comment by jakwinkler** [comment]
@orlangur I think these issues are not related.
As for your message: community members don't have access to closed-source repositories. This can be done only by Adobe team. 

I just look at the oldest issues from https://forger.magento-opensource.com/ (released few days ago).
That tool was created to help clean things up with this repos, pull request, issues and so on.

Can you pass it to proper people at Adobe to fix it?
Open Source repository should not contain any code related to commerce/paid modules.

**Comment by jakwinkler** [comment]
It would be better if those files were moved to the Commerce modules where they belong. That way, the open source version stays clean and doesn’t include code for features it doesn’t have.

**Comment by engcom-Bravo** [comment]
Hi @clockworkgeek,

Adobe Commerce Engineering team started working on this issue. We will reach out to you if we need more information and you will get notified once the issue is fixed. Please leave comments for any further questions. 

Thank you!
